google-cloud-bigtable/lib/google/cloud/bigtable/column_family_map.rb

```diff
@@ -0,0 +1,341 @@
+# frozen_string_literal: true
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/bigtable/column_family"
+
+module Google
+  module Cloud
+    module Bigtable
+      ##
+      # An accumulator for modifications to a table's column families. See
+      # {Table#column_families}. Modifications will be atomically applied to
+      # the table's column families. Entries are applied in order, meaning
+      # that earlier modifications can be masked by later ones (in the case
+      # of repeated updates to the same family, for example).
+      #
+      # @example
+      #   require "google/cloud/bigtable"
+      #
+      #   bigtable = Google::Cloud::Bigtable.new
+      #
+      #   column_families = Google::Cloud::Bigtable::ColumnFamilyMap.new
+      #
+      #   column_families.add "cf1", gc_rule: Google::Cloud::Bigtable::GcRule.max_age(600)
+      #   column_families.add "cf2", gc_rule: Google::Cloud::Bigtable::GcRule.max_versions(5)
+      #
+      class ColumnFamilyMap
```

Comment:
What do you think about keeping `ColumnFamilyMap` in its original namespace: `Table::ColumnFamilyMap`?

Reply:
I moved it for 2 reasons:

1) users can now create this object, so it should be more prominent
2) this class is now at the same level and consistently located with ColumnFamily and GcRule

Reply:
Your reasons are good. However, even though the constructor was previously "private" (undocumented), just to minimize breaking changes I would prefer to keep the class in the same namespace. That said, breaking changes are a judgement call, this is just my personal opinion.

Reply:
I don't see how this is any different than other changes like replacing protobuf GcRule objects returned from the API with the veneer GcRule objects. If users don't inspect the object, they won't know any difference.